intermediate-python-debugging/vs-code-debugging.md

```diff
@@ -0,0 +1,190 @@
+
+You can then run the debugger at any time with the green triangle play button.
+
+![Debugger play button](../assets/vs-code-debugger/debugger-play.png)
+
+Notice that the program will pause at the 1st breakpoint it encounters.
+
+**What are the values of the variables?**  
+
+### Watches
+
+You can create a watch by clicking on the `+` icon and enter any expression.
+
+![Entering a Watch](../assets/vs-code-debugger/watch-price.png)
+
+![Watches](../assets/vs-code-debugger/watch.png)
+
+
+**In the watch box, enter `number < guess`**  What does it evaluate to?  Try a few more expressions.
+
+![Watch example](../assets/vs-code-debugger/watch_example.png)
+
+### Stepping Over & Stepping Into
+
+Notice when the debugger is paused at a breakpoint you get the menu below.
+
+![Debugger Menu](../assets/vs-code-debugger/debugger-menu.png)
+
+This menu has the following buttons:
+
+- **Resume** - Continues the debugger executing until it gets to the next breakpoint, or the program exits.
+- **Step Over** - This button executes the current line and moves to the next line.  It will execute any function invoked on the current line, without pausing inside that function.
+- **Step Into** - This button goes into any function invoked on the current line and pauses at the 1st line.
+- **Step Out** - This button finishes the rest of the current function and pauses at the calling function.
+- **Restart** - This button restarts the debugger from the start.
+- **Stop** - This button stops the debugger.
+
+These buttons let you deliberately navigate through the program's execution allowing you to examine the state of the application at any step in the process.
+
+**Try using the buttons to navigate through the code.**  Do you see the values of the variables changing?
+
+<!-- >>>>>>>>>>>>>>>>>>>>>> BEGIN CHALLENGE >>>>>>>>>>>>>>>>>>>>>> -->
+<!-- Replace everything in square brackets [] and remove brackets  -->
+
+### !challenge
+
+* type: short-answer
+* id: c3eb6210-5cbd-4507-b0a5-41db00b43fb5
+* title: Step over and Step into why no difference?
+* points: [1] (optional, the number of points for scoring as a checkpoint)
+<!-- * topics: [python, pandas] (optional the topics for analyzing points) -->
+
+##### !question
+
+In the code try using the step over and step into buttons, do you notice a difference?  No, why not?
+##### !end-question
+
+##### !placeholder
+
+Why don't we see a difference?
+
+##### !end-placeholder
+
+##### !answer
+
+/.+/
+
+##### !end-answer
+
+<!-- other optional sections -->
+<!-- !hint - !end-hint (markdown, hidden, students click to view) -->
+<!-- !rubric - !end-rubric (markdown, instructors can see while scoring a checkpoint) -->
+##### !explanation 
+
+It's because we don't have any functions that we have written and used  The debugger is wisely deciding not to jump into the source code for the `input` function.
+
+##### !end-explanation
+
+### !end-challenge
+
+<!-- ======================= END CHALLENGE ======================= -->
```

Comment:
Personally, I don't think this is an effective question for students to spend their time on.

The student takeaway should be that there **is** a difference between step over and step in buttons. With this question, the student must put effort into understanding this exception. Since this is their first time with the subject, the exception may stick out more clearly in their head than the rule, which isn't ideal IMO.

Personally, I'd either cut this question, or turn this question into assessing their knowledge of something more "typical" (and possibly move into a Check for Understanding section)